fdbclient/BlobGranuleFiles.cpp

```diff
@@ -287,18 +300,28 @@ struct IndexBlockRef {
 			TraceEvent(SevDebug, "IndexBlockEncrypt_Before").detail("Chksum", chksum);
 		}
 
+		Value serializedBuff = ObjectWriter::toValue(block, IncludeVersion(ProtocolVersion::withBlobGranuleFile()));
 		EncryptBlobCipherAes265Ctr encryptor(
 		    eKeys.textCipherKey,
 		    eKeys.headerCipherKey,
 		    cipherKeysCtx.ivRef.begin(),
 		    AES_256_IV_LENGTH,
 		    getEncryptAuthTokenMode(EncryptAuthTokenMode::ENCRYPT_HEADER_AUTH_TOKEN_MODE_SINGLE),
 		    BlobCipherMetrics::BLOB_GRANULE);
-		Value serializedBuff = ObjectWriter::toValue(block, IncludeVersion(ProtocolVersion::withBlobGranuleFile()));
-		BlobCipherEncryptHeader header;
-		buffer = encryptor.encrypt(serializedBuff.contents().begin(), serializedBuff.contents().size(), &header, arena)
-		             ->toStringRef();
-		encryptHeaderRef = BlobCipherEncryptHeader::toStringRef(header, arena);
+		if (CLIENT_KNOBS->ENABLE_CONFIGURABLE_ENCRYPTION) {
+			BlobCipherEncryptHeaderRef headerRef;
+			buffer = encryptor.encrypt(
+			    serializedBuff.contents().begin(), serializedBuff.contents().size(), &headerRef, arena);
```

Comment:
Is `arena` being serialized with whatever struct that contains the `IndexBlock`? In that case, would the use of `arena` inflate the serialized size of that struct? Maybe we shall use a separate arena for the temporary encryption buffer.

Reply:
`IndexBlock` is contained within `IndexedBlobGranuleFile` struct, the `arena` used for encrypt is a client supplied arena used for serializing snapshot and/or delta file (unchanged and same as existing). The only changes is size of `EncryptHeader` StringRef, other than that there is no change to BlobGranule operations.

Reply:
Ignore what I said. I misunderstood how arena is used in serialization.